src/diagnostics.rs

```diff
@@ -199,7 +199,20 @@ fn verible_syntax(
         .arg("-")
         .spawn()
         .ok()?;
-    let re = Regex::new(r"^.+:(?P<line>.*):(?P<col>.*):\s(?P<message>.*)\s.*$").ok()?;
+
+    static RE: std::sync::OnceLock<Result<Regex, regex::Error>> = std::sync::OnceLock::new();
+    let re = RE
+        .get_or_init(|| {
+            Regex::new(
+                r"^.+:(?P<line>\d*):(?P<startcol>\d*)(?:-(?P<endcol>\d*))?:\s(?P<message>.*)\s.*$",
+            )
+            .map_err(|e| {
+                eprint!("{e}");
```

Comment:
Maybe it is better to panic here. The regex should be compiled only once and should work unless there is something outrageously wrong.